### PR TITLE
Multiversion bungeecord

### LINKED ISSUE #956: BungeeCord 1.19 support
Error on startup in new BungeeCord 1.19

```
[20:42:05 WARN]: Error loading plugin SkinsRestorer
java.lang.reflect.InvocationTargetException: null
        at jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method) ~[?:?]
        at jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:77) ~[?:?]
        at jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45) ~[?:?]
        at java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:499) ~[?:?]
        at java.lang.reflect.Constructor.newInstance(Constructor.java:480) ~[?:?]
        at net.md_5.bungee.api.plugin.PluginManager.enablePlugin(PluginManager.java:436) ~[FlameCord.jar:git:Waterfall-Bootstrap:1.19-R0.1-SNAPSHOT:7c38e1d:unknown]
        at net.md_5.bungee.api.plugin.PluginManager.loadPlugins(PluginManager.java:341) ~[FlameCord.jar:git:Waterfall-Bootstrap:1.19-R0.1-SNAPSHOT:7c38e1d:unknown]
        at net.md_5.bungee.BungeeCord.start(BungeeCord.java:299) ~[FlameCord.jar:git:Waterfall-Bootstrap:1.19-R0.1-SNAPSHOT:7c38e1d:unknown]
        at net.md_5.bungee.BungeeCordLauncher.main(BungeeCordLauncher.java:54) ~[FlameCord.jar:git:Waterfall-Bootstrap:1.19-R0.1-SNAPSHOT:7c38e1d:unknown]
        at net.md_5.bungee.Bootstrap.main(Bootstrap.java:15) ~[FlameCord.jar:git:Waterfall-Bootstrap:1.19-R0.1-SNAPSHOT:7c38e1d:unknown]
Caused by: java.lang.NoClassDefFoundError: net/md_5/bungee/connection/LoginResult$Property
        at java.lang.ClassLoader.defineClass1(Native Method) ~[?:?]
        at java.lang.ClassLoader.defineClass(ClassLoader.java:1012) ~[?:?]
        at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:150) ~[?:?]
        at net.md_5.bungee.api.plugin.PluginClassloader.findClass(PluginClassloader.java:152) ~[FlameCord.jar:git:Waterfall-Bootstrap:1.19-R0.1-SNAPSHOT:7c38e1d:unknown]
        at java.lang.ClassLoader.loadClass(ClassLoader.java:587) ~[?:?]
        at net.md_5.bungee.api.plugin.PluginClassloader.loadClass0(PluginClassloader.java:66) ~[FlameCord.jar:git:Waterfall-Bootstrap:1.19-R0.1-SNAPSHOT:7c38e1d:unknown]
        at net.md_5.bungee.api.plugin.PluginClassloader.loadClass(PluginClassloader.java:59) ~[FlameCord.jar:git:Waterfall-Bootstrap:1.19-R0.1-SNAPSHOT:7c38e1d:unknown]
        at java.lang.ClassLoader.loadClass(ClassLoader.java:520) ~[?:?]
        at net.skinsrestorer.api.property.PropertyRegistry.selectPropertyForPlatform(PropertyRegistry.java:29) ~[?:?]
        at net.skinsrestorer.shared.utils.connections.MojangAPI.<init>(MojangAPI.java:67) ~[?:?]
        at net.skinsrestorer.bungee.SkinsRestorer.<init>(SkinsRestorer.java:72) ~[?:?]
        ... 10 more
Caused by: java.lang.ClassNotFoundException: net.md_5.bungee.connection.LoginResult$Property
        at net.md_5.bungee.api.plugin.PluginClassloader.loadClass0(PluginClassloader.java:103) ~[FlameCord.jar:git:Waterfall-Bootstrap:1.19-R0.1-SNAPSHOT:7c38e1d:unknown]
        at net.md_5.bungee.api.plugin.PluginClassloader.loadClass(PluginClassloader.java:59) ~[FlameCord.jar:git:Waterfall-Bootstrap:1.19-R0.1-SNAPSHOT:7c38e1d:unknown]
        at java.lang.ClassLoader.loadClass(ClassLoader.java:520) ~[?:?]
        at java.lang.ClassLoader.defineClass1(Native Method) ~[?:?]
        at java.lang.ClassLoader.defineClass(ClassLoader.java:1012) ~[?:?]
        at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:150) ~[?:?]
        at net.md_5.bungee.api.plugin.PluginClassloader.findClass(PluginClassloader.java:152) ~[FlameCord.jar:git:Waterfall-Bootstrap:1.19-R0.1-SNAPSHOT:7c38e1d:unknown]
        at java.lang.ClassLoader.loadClass(ClassLoader.java:587) ~[?:?]
        at net.md_5.bungee.api.plugin.PluginClassloader.loadClass0(PluginClassloader.java:66) ~[FlameCord.jar:git:Waterfall-Bootstrap:1.19-R0.1-SNAPSHOT:7c38e1d:unknown]
        at net.md_5.bungee.api.plugin.PluginClassloader.loadClass(PluginClassloader.java:59) ~[FlameCord.jar:git:Waterfall-Bootstrap:1.19-R0.1-SNAPSHOT:7c38e1d:unknown]
        at java.lang.ClassLoader.loadClass(ClassLoader.java:520) ~[?:?]
        at net.skinsrestorer.api.property.PropertyRegistry.selectPropertyForPlatform(PropertyRegistry.java:29) ~[?:?]
        at net.skinsrestorer.shared.utils.connections.MojangAPI.<init>(MojangAPI.java:67) ~[?:?]
        at net.skinsrestorer.bungee.SkinsRestorer.<init>(SkinsRestorer.java:72) ~[?:?]
        ... 10 more
```

### PR DESCRIPTION
**Pull request by AlexProgrammerDE**
Had to create a new submodule architecture to make this work, but it's done!

### CODE REVIEW

File: api/src/main/java/net/skinsrestorer/api/reflection/ReflectionUtil.java [hunk 1]
```diff
@@ -65,6 +65,15 @@ public class ReflectionUtil {
     private ReflectionUtil() {
     }
 
+    public static boolean classExists(String clazz) {
```
Comment (xknat) [inline api/src/main/java/net/skinsrestorer/api/reflection/ReflectionUtil.java:68]: Consider using a better name
e.g. classStringName

**Review by xknat — CHANGES_REQUESTED** [review]
- [x] consider adding notes why & when BungeePropertyNew is used

File: api/src/main/java/net/skinsrestorer/api/reflection/ReflectionUtil.java (hunk 1 above)
Comment (AlexProgrammerDE) [reply api/src/main/java/net/skinsrestorer/api/reflection/ReflectionUtil.java:68]: Does not make sense to change, as all other method parameters in this class use clazz. It's just shorter. And "class" is a reserved java keyword.

**Review by xknat — APPROVED** [review]
LGTM, thank you for doing this!

**Pull request merged**